sw/tools/generators/gen_airframe.ml

```diff
@@ -337,9 +337,17 @@ let parse_ap_only_commands = fun out ap_only ->
 
 let parse_command = fun out command no ->
   let command_name = "COMMAND_"^ExtXml.attrib command "name" in
-  define_out out command_name (string_of_int no);
   let failsafe_value = int_of_string (ExtXml.attrib command "failsafe_value") in
-  { failsafe_value = failsafe_value; foo = 0}
+  let actuator_type = ExtXml.attrib_or_default command "actuator_type" "REAL" in 
+  define_out out command_name (string_of_int no);
+  { failsafe_value = failsafe_value; foo = 0 ; actuator_type = actuator_type }  
+
+let count_commands_by_type commands_params =
+  List.fold_left (fun acc cmd ->
+    let subtype = cmd.actuator_type in
+    let count = try List.assoc subtype acc with Not_found -> 0 in
+    (subtype, count + 1) :: (List.remove_assoc subtype acc)
+  ) [("REAL", 0); ("VIRTUAL", 0); ("PASSIVE", 0); ("OTHER", 0)] commands_params
```

Comment:
Is it really necessary to have default categories. The list is not even exhaustive since it should work with any name. I suggest that the initial list is just empty. If the actuator is not defining a type, the name `REAL` will be used by default and only this type will be generated. If a controller needs more, it is up to the user to define more in the airframe file.